packages/clerk-js/src/core/state.ts

```diff
@@ -0,0 +1,20 @@
+import type { State as StateInterface } from '@clerk/types';
+import { computed, effect } from 'alien-signals';
+
+import { eventBus, type SignInUpdatePayload } from './events';
+import { signInSignal } from './signals';
+
+export class State implements StateInterface {
+  signInSignal = signInSignal;
+
+  __internal_effect = effect;
+  __internal_computed = computed;
+
+  constructor() {
+    eventBus.on('signin:update', this.onSignInUpdated);
+  }
+
+  private onSignInUpdated = (payload: SignInUpdatePayload) => {
+    this.signInSignal({ resource: payload.resource });
+  };
```

Comment:
Similarly to my previous comments, this looks like it can be made generic & more resource-agnostic. I am okay with this as the first iteration, but adding a 2nd resource in the same manner should be a good signal (pun intended) that we need to refactor this